exporter/elasticsearchexporter/metric_grouping.go

```diff
@@ -35,30 +39,59 @@ func newDataPointHasher(mode MappingMode) dataPointHasher {
 // TODO use https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/internal/exp/metrics/identity
 
 type (
-	ecsDataPointHasher  struct{}
+	// ecsDataPointHasher solely relies on hashCombined because attributes at resource and data point may overwrite one another.
+	ecsDataPointHasher struct{}
+	// otelDataPointHasher does not use hashCombined as resource, scope and data points are independent.
 	otelDataPointHasher struct{}
 )
 
-func (h ecsDataPointHasher) hashDataPoint(resource pcommon.Resource, _ pcommon.InstrumentationScope, dp datapoints.DataPoint) uint32 {
-	hasher := fnv.New32a()
+func (h ecsDataPointHasher) hashResource(_ pcommon.Resource) uint32 {
+	return 0
+}
+
+func (h ecsDataPointHasher) hashScope(_ pcommon.InstrumentationScope) uint32 {
+	return 0
+}
+
+func (h ecsDataPointHasher) hashDataPoint(_ datapoints.DataPoint) uint32 {
+	return 0
+}
 
-	mapHashExcludeReservedAttrs(hasher, resource.Attributes())
+func (h ecsDataPointHasher) hashCombined(resource pcommon.Resource, _ pcommon.InstrumentationScope, dp datapoints.DataPoint) uint32 {
+	merged := pcommon.NewMap()
+	merged.EnsureCapacity(resource.Attributes().Len() + dp.Attributes().Len())
+	resource.Attributes().CopyTo(merged)
+	// scope attributes are ignored in ECS mode
+	for k, v := range dp.Attributes().All() {
+		_ = merged.PutEmpty(k).FromRaw(v.AsRaw())
```

Comment:
[to reviewer] There's no generic `map.Put` function and I'm reluctant to add a per-type switch case here, as it will not support new types if there's one in the future.